### PR TITLE
Adds integration tests for the geoip processor

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

This PR adds integration tests for the geoip processor.

I cherry picked the original commit from @asifsmohammed 's PR #4173

```
git cherry-pick d7cd8f2207c07fbea4a0b39f4784c01a9d1fb6db
```

Then I fixed the tests with a few improvements:

* Get the GeoIP service by using existing code as much as possible.
* Download the test databases to the `build` directory so that it cleans with `./gradlew clean`.
* Dynamically get the `data-prepper.dir` when generating the path to keep different runs from conflicting
 
### Issues Resolved

N/A
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/geoip-processor/src/test/java/org/opensearch/dataprepper/plugins/geoip/extension/MaxMindConfigTest.java [hunk 1]
```diff
@@ -14,38 +14,41 @@
...
 
-    @BeforeEach
-    void setup() {
-        maxMindConfig = new MaxMindConfig();
+    private MaxMindConfig createObjectUnderTest() {
+        return new MaxMindConfig();
     }
 
-    @Test
+    @RepeatedTest(2)
```
Comment (chenqi0805) [inline data-prepper-plugins/geoip-processor/src/test/java/org/opensearch/dataprepper/plugins/geoip/extension/MaxMindConfigTest.java:37]: QUES: why are we repeating running this test case?

File: data-prepper-plugins/geoip-processor/src/integrationTest/java/org/opensearch/dataprepper/plugins/geoip/processor/GeoIPProcessorIT.java [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+                "              country: " + COUNTRY_URL + "\n" +
+                "              city: " + CITY_URL + "\n" +
+                "              asn: " + ASN_URL + "\n" +
+                "            cache_count: 1024\n" +
+                "            database_refresh_interval: P3D";
+
+        geoipProcessorConfig = OBJECT_MAPPER.readValue(geoipProcessorConfigYaml, GeoIPProcessorConfig.class);
+        GeoIpServiceConfig geoipServiceConfig = OBJECT_MAPPER.readValue(geoipServiceConfigYaml, GeoIpServiceConfig.class);
+
+        System.setProperty("data-prepper.dir", "build");
```
Comment (oeyh) [inline data-prepper-plugins/geoip-processor/src/integrationTest/java/org/opensearch/dataprepper/plugins/geoip/processor/GeoIPProcessorIT.java:96]: This seems to overwrite the same system property set in build.gradle. But both places point to the same directory, so there is probably no issue.

File: data-prepper-plugins/geoip-processor/src/test/java/org/opensearch/dataprepper/plugins/geoip/extension/MaxMindConfigTest.java (hunk 1 above)
Comment (dlvenable) [reply data-prepper-plugins/geoip-processor/src/test/java/org/opensearch/dataprepper/plugins/geoip/extension/MaxMindConfigTest.java:37]: Running this test twice ensures that the `System.getProperty` call is made dynamically.

This code will evaluate this once and then make it a constant:

https://github.com/opensearch-project/data-prepper/blob/67650987919f20aba06e22d284d1e66d38f42be9/data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/geoip/extension/MaxMindConfig.java#L28

So this test passes when run once with the above code. If you run it twice, then you are sure it is getting the property dynamically.

File: data-prepper-plugins/geoip-processor/src/integrationTest/java/org/opensearch/dataprepper/plugins/geoip/processor/GeoIPProcessorIT.java (hunk 2 above)
Comment (dlvenable) [reply data-prepper-plugins/geoip-processor/src/integrationTest/java/org/opensearch/dataprepper/plugins/geoip/processor/GeoIPProcessorIT.java:96]: I originally had this line and then decided to set it in the `build.gradle`. So this line should be removed. Thanks!

Comment (dlvenable) [reply data-prepper-plugins/geoip-processor/src/integrationTest/java/org/opensearch/dataprepper/plugins/geoip/processor/GeoIPProcessorIT.java:96]: Actually, leaving this here will also help if you run the tests from IntelliJ directly. So I will leave it as-is.

**Review by chenqi0805 — APPROVED** [review]

**Comment by chenqi0805** [comment]
Just a question. Should #4173 get closed once this merged?

**Comment by dlvenable** [comment]
> Just a question. Should #4173 get closed once this merged?

Yes, I will close that issue after we merge this.

**Review by oeyh — APPROVED** [review]

**Pull request merged**